docs/user_guide/usage_patterns/data_transfer/detecting_data_access/index.rst

```diff
@@ -0,0 +1,161 @@
+.. currentmodule:: globus_sdk
+
+.. _userguide_detecting_data_access:
+
+Detecting data_access
+=====================
+
+Globus Collections come in several varieties, but only some of them have a
+``data_access`` scope.
+
+``data_access`` scopes control application access to collections, allowing
+users to revoke access for an application independent from other application
+permissions.
+Revoking consent stops data transfers and other operations.
+
+Because only some collection types have ``data_access`` scopes, application
+authors interacting with these collections may need to detect the type of
+collection and determine whether or not the scope will be needed.
+
+For readers who prefer to start with complete working examples, jump ahead to the
+:ref:`example script <userguide_detecting_data_access_example>`.
+
+Accessing Collections in Globus Transfer
+----------------------------------------
+
+The Globus Transfer service acts as a central registration hub for collections.
+Therefore, in order to get information about an unknown collection, we will
+need a :class:`TransferClient` with credentials.
+
+The following snippet creates a client and uses it to fetch a collection from
+the service:
+
+.. code-block:: python
+
+    import globus_sdk
+
+    # Tutorial Client ID - <replace this with your own client>
+    NATIVE_CLIENT_ID = "61338d24-54d5-408f-a10d-66c06b59f6d2"
+    USER_APP = globus_sdk.UserApp("detect-data-access-example", client_id=NATIVE_CLIENT_ID)
+
+    # Globus Tutorial Collection 1
+    # https://app.globus.org/file-manager/collections/6c54cade-bde5-45c1-bdea-f4bd71dba2cc
+    # replace with your own COLLECTION_ID
+    COLLECTION_ID = "6c54cade-bde5-45c1-bdea-f4bd71dba2cc"
+
+    transfer_client = globus_sdk.TransferClient(app=USER_APP)
+
+    collection_doc = transfer_client.get_endpoint(COLLECTION_ID)
+
+.. caution::
+
+    Careful readers may note that we the :meth:`TransferClient.get_endpoint`
+    method to lookup a collection.
+
+    The Transfer service contains both Endpoints and Collections, and for
+    historical reasons both document types are available from the Get Endpoint
+    API.
+    This is the correct API to use not only when looking for information about
+    a Collection, but also for detecting whether an ID belongs to a Collection
+    or Endpoint, in cases where it is ambiguous.
+
+
+Reading Collection Type
+-----------------------
+
+There are two attributes we need from the collection document to determine
+whether or not a ``data_access`` scope is used.
+
+First, whether or not the collection is a GCSv5 Mapped Collection:
+
+.. code-block:: python
+
+    entity_type: str = collection_doc["entity_type"]
+    is_v5_mapped_collection: bool = entity_type == "GCSv5_mapped_collection"
```

Comment:
Let's rip out the type annotations, which don't help with readability IMHO.

```suggestion
    entity_type = collection_doc["entity_type"]
    is_v5_mapped_collection = entity_type == "GCSv5_mapped_collection"
```

Reply:
Although in this case I think it's clear what the types are from context (and will take the suggestion to remove), I think annotating our assignments may help users understand payload shapes in some cases. That _assumes_ (and, IMO, this is a big assumption, but one that I have been making) that type-annotated Python has proliferated to the point that our users are likely to be familiar with the syntax.

Reply:
For a complete script, that may be the case but for code snippet that converts from an English sentence to a Python line I think it may get in the way.

I don't know for a fact that it's adding to -- nor whether it's harming -- readability, but I think it _might_ be obscuring the simple English->Python value that these inline code snippets offer.